File: arrays/summary_ranges.rb
# @param {Integer[]} nums
# @return {String[]}
def summary_ranges(nums)
    ind = 0
    res = []

    nums.each_index do |i|
        next if nums[i + 1] == nums[i] + 1

        res << (ind == i ? nums[i].to_s : "#{nums[ind]}->#{nums[i]}")
        ind = i + 1
    end

    res
end

File: arrays/summary_ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums:
            return []

        start = 0
        res = []

        for i in range(len(nums)):
            if i + 1 < len(nums) and nums[i + 1] == nums[i] + 1:
                continue

            if start == i:
                result.append(str(nums[i]))
            else:
                result.append(f"{nums[start]}->{nums[i]}")
            
            start = i + 1

        return res